### ISSUE TITLE
Language Filter

### ISSUE THREAD

**Issue by Made1908** [sleeyax/stremio-easynews-addon#29]
It would be nice to have a language Filter, so that only specified languages are shown.

**Comment by sleeyax** [comment]
Not sure if that's possible, I don't think Easynews includes language info in search results but I'll have to double check. Maybe it can be extracted from the title (similar to how quality/resolution is currently extracted) but it definitely won't be a perfect solution as many titles don't have a language code in them.

**Comment by Screamio** [comment]
Language filters would be really helpful indeed.

Probably in form of a Regex search filter as an optional filter which could be set?
Could be a new sorting option, which the user could configure and it would allow to list matching file names at the beginning of the list.

**Comment by mikmc55** [comment]
the code is here open source to test with any regex filtering you want, its not possible from the easynews console itself so not straightforward to implement

**Comment by linsetlogan** [comment]
If you are not english, it's a nighmare without lang filter.

**Comment by kibabyte** [comment]
Would it be possible to add a field in the configuration that simply appends something to the search? Appending spanish, french, german or whatever will probably already yield quite usable results. Please excuse my ignorance if this is not how it would work.

**Comment by sleeyax** [comment]
> Would it be possible to add a field in the configuration that simply appends something to the search? Appending spanish, french, german or whatever will probably already yield quite usable results. Please excuse my ignorance if this is not how it would work.

Yes, that could work as a work-around if no other more robust solution is found.

**Comment by kibabyte** [comment]
Was a better solution found here? I will try to implement what I am wanting to do when there is some nice deal on easynews for black friday, although a slightly less janky solution would surely be appreciated.

**Comment by newspeer** [comment]
That would be great!

**Comment by kingunlim** [comment]
maybe something in the pipes? 😎

**Comment by kingunlim** [comment]
maybe someone else knows a working solution to the language filter problem?

**Comment by kibabyte** [comment]
> maybe someone else knows a working solution to the language filter problem?

Only way I can think of is self hosting the add on with a modified search prompt. Might aswell make it adjustable to instead of hardcoding it in. I will look into it in the next few weeks maybe.

**Comment by Marvins13** [comment]
A simple fix I made for myself was to add a filter with "german" to the For loop of the defineStreamHandler and the defineMetaHandler in the addon.ts. 
And now I have it running myself with a cloudflare worker.

`if (!title.toUpperCase().includes("GERMAN")) {
          continue;
        }`

**Comment by newspeer** [comment]
Thanks, unfortunately doesn't work for me. Easynews doesn't return anything for me. Tested with many current German releases.

`        if (isBadVideo(file) || !matchesTitle(title, search, false)) {
          continue;
        }
      if (!title.toUpperCase().includes("GERMAN")) { continue; }

        videos.push({
          id: `${prefix}${file.sig}`,`

and
`        if (isBadVideo(file)) {
          continue;
        }
if (!title.toUpperCase().includes("GERMAN")) { continue; }
        // For series there are multiple possible queries that could match the title.
        // We check if at least one of them matches.
        if (type === 'series') {
          const queries = [`

Did I place the filter wrong?

**Comment by sleeyax** [comment]
Create a PR
